src/@next/AuthenticationInput/AuthenticationInput.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { ClipboardEvent, RefObject, useEffect, useState } from 'react';
+import { StyledInputsContainer } from './AuthenticationInputStyle';
+import { AuthenticationInputSingle } from './AuthenticationInputSingle';
+
+export interface AuthenticationInputProps
+  extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'> {
+  numberOfInputs: number;
+  disabled?: boolean;
+  error?: boolean;
+  forwardedRef?: RefObject<HTMLDivElement>;
+  onChange?: (value: string | null) => void;
+}
+const AuthenticationInputComponent = ({
+  numberOfInputs = 6,
+  error,
+  onChange,
+  forwardedRef,
+  ...rest
+}: AuthenticationInputProps) => {
+  const [inputValues, setInputValues] = useState(
+    new Array<number | ''>(numberOfInputs).fill('')
+  );
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const inputArray = new Array(numberOfInputs).fill('');
+
+  useEffect(() => {
+    if (inputValues.some(v => !Number.isInteger(v))) {
+      onChange?.(null);
+      return;
+    }
+    onChange?.(inputValues.join(''));
+  }, [inputValues, onChange]);
+
+  const handleOnChange = (value: number, index: number) => {
+    if (!Number.isInteger(value)) {
+      const newInputValues = [...inputValues];
+      newInputValues[index] = '';
+      setInputValues(newInputValues);
+      return;
+    }
+    const newInputValues = [...inputValues];
+    newInputValues[index] = value;
+    setInputValues(newInputValues);
+    const nextInputIndex = index + 1;
+
+    if (nextInputIndex <= numberOfInputs - 1) {
+      setCurrentIndex(nextInputIndex);
+    }
+  };
+
+  const handlePaste = (event: ClipboardEvent) => {
```

Comment:
suggest to use a useCallback here

Reply:
I don't think we need that right? Doesn't seem to be expensive operation

Reply:
the function seems to get created on every render unnecessarily if we don't have any deps with the props in this function and it does not need to recreate, i think its best we can wrap in a callback, but i am fine with keeping it as it is too, we can optimize incase of any issues

Reply:
Hmm useCallback and useMemo actually has a downside, unless it's an expensive operation the rerender really doesn't make a difference, it only make it hard to read. Here is an interesting take: https://kentcdodds.com/blog/usememo-and-usecallback